### PR TITLE
Fixes #16922 - Clean old tasks automaticaly

### PR DESCRIPTION
**Pull request by adamruzicka**
This PR includes https://github.com/theforeman/foreman-tasks/pull/247. Based on discussion in original PR this now also allows to distinguish between green an red (and any other) tasks and set different cleanup intervals.

### CODE REVIEW

File: config/foreman-tasks.yaml.example [hunk 1]
```diff
@@ -16,11 +16,24 @@
   :cleanup:
 #
 # the period after which to delete all the tasks (by default all tasks are not being deleted after some period)
+# will be deprecated (TODO: mention when) and the use of rules is recommended
```
Comment (iNecas) [inline config/foreman-tasks.yaml.example]: Whatever comes with Foreman 1.18?

File: lib/foreman_tasks/cleaner.rb [hunk 2]
```diff
@@ -11,11 +11,17 @@ def self.run(options)
           end
         end
         if cleanup_settings[:after]
+          # TODO: Mention version in which this will be removed completely
+          ActiveSupport::Deprecation.warn('You are using legacy interface, please migrate from using :after to cleanup rules')
```
Comment (iNecas) [inline lib/foreman_tasks/cleaner.rb]:
```ruby
Foreman::Deprecation.deprecation_warning("1.18", _(':after setting in tasks cleanup section is deprecated, use :after in :rules section to set the value. to cleanup rules'))
```

**Comment by iNecas** [comment]
Let's rebase this and add some tests before we proceed

**Comment by adamruzicka** [comment]
The test matrix is all green but the build was marked as failed. Let's try it again

[test]

File: config/foreman-tasks.yaml.example [hunk 3]
```diff
@@ -13,14 +13,27 @@
...
+#
+# Rules defined in this section by default don't operate
+# on tasks specified in the actions section. This behavior
+# can be overriden by setting the override_actions to true
+   :rules:
+     # Delete successful tasks after a month
+     - :filter: result = success
+       :after: 30d
+     # Delete everything (any action, any state) after one year
+     - :states: []
```
Comment (iNecas) [inline config/foreman-tasks.yaml.example]: I know I just keep adding comments here, but this is the last one (hopefully:). It's not very clear that empty states means all while missing/nil means 'stopped'. What about introducing 'all' as recognized value that would replace the `[]` optoin we have now.

**Comment by iNecas** [comment]
Looks good, waiting for tests to finish. There is one failure so far, but seems not to be related directly

**Comment by adamruzicka** [comment]
The tests are done, with one (most likely) unrelated failure.

**Comment by iNecas** [comment]
Thanks @adamruzicka and @mbacovsky

**Pull request merged**